### PR TITLE
Feat: editable safeTxGas in Advanced details

### LINKED ISSUE #2122: [TX flow] safeTxGas
Existing transactions already have it but for newly created transactions:

* Add "Advanced details" to the "Transaction details" accordion
* Show safeTxGas and Raw data in it
* Make safeTxGas editable

Screenshot 2023-06-19 at 11 06 49

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

Resolves #2122

## How this PR fixes it
Adds a `<PartialSummary>` component to display Advanced Details for newly created txs.
It contains the `safeTxGas` and the raw data of a SafeTransaction. `safeTxGas` is editable.

## Screenshots
Screenshot 2023-06-20 at 11 39 48
Screenshot 2023-06-20 at 11 39 43
Screenshot 2023-06-20 at 11 39 33

### CODE REVIEW

**Comment by usame-algan** [comment]
Is it possible to make the input even smaller so there is less of a jump?

**Comment by katspaugh** [comment]
The best would be to open it in in a popover, so that it doesn't affect the layout at all, but I think it's not worth spending more time on this.

File: src/components/transactions/TxDetails/SafeTxGasForm.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  const [editing, setEditing] = useState(false)
+
+  const formMethods = useForm<{ safeTxGas: number }>({
+    defaultValues: {
+      safeTxGas,
+    },
+    mode: 'onChange',
+  })
+
+  const onSubmit = (values: { safeTxGas: number }) => {
```
Comment (iamacook) [inline src/components/transactions/TxDetails/SafeTxGasForm.tsx:18]: Nit: let's extract the type so we needn't define it twice.

File: src/components/transactions/TxDetails/SafeTxGasForm.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                  valueAsNumber: true,
+                  min: 0,
+                  setValueAs: Math.round,
+                })}
+              />
+              <Button type="submit" size="small" variant="contained" sx={{ ml: 1 }}>
+                Save
+              </Button>
+            </form>
+          </FormProvider>
```
Comment (iamacook) [inline src/components/transactions/TxDetails/SafeTxGasForm.tsx:56]: We don't need the `FormProvider` here and I'd suggest using our `NumberField` instead.

```suggestion
          <form onSubmit={formMethods.handleSubmit(onSubmit)} style={{ display: 'flex' }}>
            <NumberField
              size="small"
              autoFocus
              type="number"
              error={!!formMethods.formState.errors.safeTxGas}
              sx={{ width: '7em' }}
              {...formMethods.register('safeTxGas', {
                valueAsNumber: true,
                min: 0,
                setValueAs: Math.round,
              })}
            />
            <Button type="submit" size="small" variant="contained" sx={{ ml: 1 }}>
              Save
            </Button>
          </form>
```

**Review by iamacook — COMMENTED** [review]
Not overtly necessary but I noticed the following as well:

- The `Popover` `background-color` does not quite match that of the `DecodedTx` one.
- There is a small pixel shift of the amount when the `Popover` opens.

safetxgas

**Review by iamacook — APPROVED** [review]
Looks good to me 👍 

image

**Pull request merged**